onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/reduction_op_builder.cc

```diff
@@ -180,13 +177,23 @@ bool ReductionOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializ
   }
 
   if (op == "ReduceMean") {
+    const bool noop_with_empty_axes = helper.Get("noop_with_empty_axes", 0) != 0;
     if (inputs.size() > 1 && inputs[1].node_arg.Exists()) {
       const auto& axes_name = inputs[1].node_arg.Name();
       if (!Contains(initializers, axes_name)) {
         LOGS_DEFAULT(VERBOSE) << "Axes of ReduceMean must be a constant initializer.";
         return false;
       }
     }
+    // Note: For the case - ReduceMean 18+ with noop_with_empty_axes attribute set as 1,
+    // currently we hit an issue in NNAPI where it does not allow adding an operand as both an input and output.
+    // This issue may arise from handling no-ops like Identity and ReduceX with noop_with_empty_axes set.
+    // TODO: Support the case when a more complete solution is available.
```

Comment:
Out of interest could we do a Reshape with the same shape as a fake no-op with minimal cost?

Reply:
good point. I guess adding a Reshape with same shape as the no-op could be an idea that can work. 

but not sure if it will bring about any overhead as to model size / also the context switch between nnapi cpu impl vs hardware accelerator impl since Reshape does not run on hardware accelerator on some cpus. (I saw previously we have some handlings for that.

Reply:
Ah ok. If Reshape is CPU that's not helping. Multiply by 1 with broadcasting may be another option (i.e. Mul with the other input being a single 1)